### PR TITLE
pd_client: add some function to buckets

### LINKED ISSUE #14240: `BucketStat` needs more power
## Development Task
There are many code called `merge_bucket_stats` and `new_bucket_stats`:
https://github.com/tikv/tikv/blob/c5ce165ad35121114892fb7b27aeeaae3c3e7afc/components/raftstore/src/store/worker/split_controller.rs#L459-L466

### PR DESCRIPTION
**Pull request by bufferflies**
### What is changed and how it works?

Issue Number: Close #14240

What's Changed:

```commit-message
1. add `from_meta` constructor 
2. add `merge` to merge delta flow
3. add `add_flow` to add flow for given key range 
```

### Check List 

Tests 

- No code

Side effects

### Release note 

```release-note
None
```

### CODE REVIEW

File: components/raftstore/src/store/fsm/apply.rs [hunk 1]
```diff
@@ -3941,12 +3941,8 @@ where
...
         self.delegate.term = apply.term;
         if let Some(meta) = apply.bucket_meta.clone() {
-            let buckets = self
-                .delegate
-                .buckets
-                .get_or_insert_with(BucketStat::default);
-            buckets.stats = new_bucket_stats(&meta);
-            buckets.meta = meta;
+            let new = BucketStat::from_meta(meta);
+            self.delegate.buckets.replace(new);
```
Comment (SpadeA-Tang) [inline components/raftstore/src/store/fsm/apply.rs:3945]: This will refresh the creation time of the buckets.

Comment (bufferflies) [reply components/raftstore/src/store/fsm/apply.rs:3945]: Good catch, the `create_time` is useless, I will remove it.

Comment (SpadeA-Tang) [reply components/raftstore/src/store/fsm/apply.rs:3945]: why?

Comment (bufferflies) [reply components/raftstore/src/store/fsm/apply.rs]: yes, I have added new function to modify meta and stats.

**Review by SpadeA-Tang — APPROVED** [review]

**Review by BusyJay — APPROVED** [review]

**Review by iosmanthus — APPROVED** [review]

**Pull request merged**